jormungandr/src/metrics/backends/simple_counter.rs

```diff
@@ -101,45 +101,46 @@ impl Default for SimpleCounter {
 
 impl MetricsBackend for SimpleCounter {
     fn add_tx_recv_cnt(&self, count: usize) {
-        self.tx_recv_cnt.fetch_add(count, Ordering::SeqCst);
+        self.tx_recv_cnt.fetch_add(count, Ordering::Relaxed);
```

Comment:
Doing this for `load`/`store` (i.e. values updated from another reliably source) is all right, but could `fetch_add` inaccurately increment the count in this mode?

Reply:
Hm, could you give an example? I don't really understand how this can happen.

Reply:
If two threads do this concurrently, both of them can load the same value and then store the value + 1. This needs to order the load half after the store half of any other concurrent increment. I'd use `Ordering::AcqRel` (we don't care about an absolute ordering of the increments, so `SeqCst` is too heavy indeed). I mean, we're splitting hairs here, but let's split them correctly :trollface:

Reply:
I stand corrected by [the nomicon](https://doc.rust-lang.org/nomicon/atomics.html#relaxed).